### DISCUSSION TITLE
Get type of Prisma validator

### DISCUSSION THREAD

**Discussion by IRediTOTO** [prisma/prisma#14800] [Q&A] [answered]
Hi, I read about Prisma validator https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/prisma-validator
But can we extends it more? 
```
const userEmail = Prisma.validator<Prisma.UserSelect>()({
  email: true,
})

const user = await prisma.user.findUnique({
  where: {
    id: 3,
  },
  select: userEmail,
})
...
In other file
const MyComponent: FC<data: Prisma.infer<typeof userEmail>> = ()=>{}  // Can we do like this?
I hope this can equal
const MyComponent: FC<data: ReturnType<typeof user> = ()=>{}  

```
Why do we want this? Just because the userEmail can be used in many function, I don't want this place call `typeof user`  or `typeof updateUser`  they have same return data by `userEmail `  select

**Comment by pantharshit00** [comment, marked as answer]
I like the idea. It would be better if you can [make a feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for this as discussion will get lost and won't be tracked for feature updates.

**Comment by julienben** [comment]
This is how I do it:
```ts
const findFirst = async () => await prisma.prediction.findFirst({ select: defaultPredictionSelect });

export type TDefaultPrediction = inferAsyncReturnType<typeof findFirst>;
```
It works but it's obviously not the most elegant thing in the world. Is there a better option that I'm not seeing?

**Reply by julienben** [reply]
Ok found the real solution in [this blog post](https://www.prisma.io/blog/satisfies-operator-ur8ys8ccq7zb#infer-the-output-type-of-methods-like-findmany-and-create):
```ts
export type TDefaultPrediction = Prisma.PredictionGetPayload<{
  select: typeof defaultPredictionSelect;
}>;
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
